src/main/java/com/cs56fitnessapp/models/Meal.java

```diff
@@ -1,29 +1,91 @@
 package com.cs56fitnessapp.models;
+import java.util.ArrayList;
+// import java.util.Date;
+import java.time.LocalDate;
+
 
 /**
- * @author Nerissa Hsieh
+ * @author Nerissa Hsieh & Jimwell Castillo
  * Date Created: 10/03/17
- * Date Modified: 10/03/17
+ * Date Modified: 10/20/17
  */
 
-public class Meal implements DietFacts {
+public class Meal implements DietFacts{
+
+      //date
+    private LocalDate date;
+    private String title;
+    //foodList
+    private ArrayList <FoodEntry> foodList;
+    private String MealTitle;
+    private int MealCalories;
+    private int MealFat;
+    private int MealCarb;
+    private int MealProtein;
+
+
+    public Meal (LocalDate date, String MealTitle, ArrayList < FoodEntry > foodList) {
+        this.date = date;
+        this.title = MealTitle;
+        this.foodList = new ArrayList<>();
+     }
+
+    /**Edited 10.20.17, Jim
+     * **/
+
     @Override
     public int getCaloriesIn() {
-        return 0;
-    }
 
-    @Override
-    public double getTotalFatIn() {
-        return 0;
-    }
+            for(int i = 0; i<this.foodList.size(); i++){
+                FoodEntry food = this.foodList.get(i);
+                if(food.getCalories() == MealCalories){
+                    return i;
+                }
+            }
+            return 0;
 
```

Comment:
Method getCaloriesIn() like all subsequent overridden methods will always return 0. 
Inside the method if condition would be never met. 
If we imagine it will, for the list of size 5, the method will return values from 0 to 4. 
So no matter what food I ate during lunch it will contain 0, 1, 2, 3 or 4 calories.

Logic is not correct.